### PR TITLE
feat: Add checksums for files

### PR DESCRIPTION
**Pull request by ArneGudermann**
This PR add the CRC32C and MD5 hash to the File skel. 

Resolves: #1176

### CODE REVIEW

**Review by sveneberth — CHANGES_REQUESTED** [review]
Nice, can you add this in [`write`](https://github.com/viur-framework/viur-core/blob/40d4e8228e72110c69d86ed80e680a55efc64656/src/viur/core/modules/file.py#L603-L633) too?

File: src/viur/core/modules/file.py [hunk 1]
```diff
@@ -352,6 +352,14 @@ class FileLeafSkel(TreeSkel):
...
         visible=False,
     )
+    crc32c_hash = StringBone(
+        descr="CRC32C hash for this File",
+        readOnly=True
+    )
+    md5_hash = StringBone(
+        descr="MD5 hash for this File",
+        readOnly=True
+    )
```
Comment (phorward) [inline src/viur/core/modules/file.py:362]: Can we rename it to checksum to clarify that this feature is used to identify same files?
```suggestion
    crc32c_checksum = StringBone(
        descr="CRC32C checksum",
        readOnly=True,
    )
    md5_checksum = StringBone(
        descr="MD5 checksum",
        readOnly=True,
    )
```

File: src/viur/core/modules/file.py [hunk 2]
```diff
@@ -621,6 +629,8 @@ def write(self, filename: str, content: t.Any, mimetype: str = "text/plain", wid
         skel["weak"] = True
         skel["width"] = width
         skel["height"] = height
+        skel["crc32c_hash"] = base64.b64decode(blob.crc32c).hex()
+        skel["md5_hash"] = base64.b64decode(blob.md5_hash).hex()
```
Comment (phorward) [inline src/viur/core/modules/file.py:633]:
```suggestion
        skel["crc32c_checksum"] = base64.b64decode(blob.crc32c).hex()
        skel["md5_checksum"] = base64.b64decode(blob.md5_hash).hex()
```

File: src/viur/core/modules/file.py [hunk 3]
```diff
@@ -907,6 +917,8 @@ def add(self, skelType: SkelType, node: db.Key | int | str | None = None, *args,
             skel["size"] = blob.size
             skel["parentrepo"] = rootNode["key"] if rootNode else None
             skel["weak"] = rootNode is None
+            skel["crc32c_hash"] = base64.b64decode(blob.crc32c).hex()
+            skel["md5_hash"] = base64.b64decode(blob.md5_hash).hex()
```
Comment (phorward) [inline src/viur/core/modules/file.py:921]:
```suggestion
            skel["crc32c_checksum"] = base64.b64decode(blob.crc32c).hex()
            skel["md5_checksum"] = base64.b64decode(blob.md5_hash).hex()
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, thanks for achieving this issue.

First of all, I would like to refer the hashed value as "checksum" as it should be an indicator for same files, regardless of the filename and the size. 

Then, I'm wondering about the chosen hash algorithms. Why did you chose CRC32c and MD5?

After a little research, I think that SHA-256 would be the best choice for collision resistance. MD5, SHA-1 and CRC32c are not designed for collision resistance. Therefore, I would prefer to only provide a sha256 checksum for all files, and avoid use of two hash algorithms (what's the advantage?). 

I'm open for discussions.

**Comment by ArneGudermann** [comment]
@phorward i used [md5](https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.blob.Blob#google_cloud_storage_blob_Blob_md5_hash) and  [crc32c](https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.blob.Blob#google_cloud_storage_blob_Blob_crc32c) because they are the only two that are provided by google storage. otherwise we would have to download the whole blob and hash it ourselves.

**Review by phorward — APPROVED** [review]
IMHO, we can use this for now.

**Review by sveneberth — APPROVED** [review]

**Pull request merged**